### PR TITLE
feat(build): Add 32bits RTE header files for x86 and arm

### PR DESCRIPTION
**Pull request by wangzihao3**
Signed-off-by: wangzihao <wangzihao18@huawei.com>

### CODE REVIEW

File: include/arch/x86/rte_atomic.h [hunk 1]
```diff
@@ -216,7 +216,7 @@ static inline int rte_atomic32_dec_and_test(rte_atomic32_t *v)
 }
 #endif
 
-#ifdef RTE_ARCH_I686
+#if defined(__i686__)
 #include "arch/x86/rte_atomic_32.h"
 #else
```
Comment (mynktl) [inline include/arch/x86/rte_atomic.h]: Better to add a check for `__x86_64__`.

**Review by mynktl — APPROVED** [review]
/lgtm

File: include/arch/x86/rte_atomic.h (hunk 1 above)
Comment (wangzihao3) [reply include/arch/x86/rte_atomic.h]:
```
#if defined(__x86_64__) || defined(__i686__)
#include <arch/x86/rte_pause.h>	#include <arch/x86/rte_pause.h>
#include <arch/x86/rte_atomic.h>	#include <arch/x86/rte_atomic.h>
```
Is it for better reading? we have checked that only ` __x86_64__` and `__i686__` could include `arch/x86/rte_atomic.h`

Comment (wangzihao3) [reply include/arch/x86/rte_atomic.h]: oh I got it, maybe other files will include `<arch/x86/rte_atomic.h>`. I will update it.

**Review by mynktl — APPROVED** [review]

File: include/arch/x86/rte_atomic_32.h [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+ */
+
+/*
+ * Inspired from FreeBSD src/sys/i386/include/atomic.h
+ * Copyright (c) 1998 Doug Rabson
+ * All rights reserved.
+ */
+
+#ifndef _RTE_ATOMIC_X86_H_
+#error do not include this file directly, use <rte_atomic.h> instead
```
Comment (pawanpraka1) [inline include/arch/x86/rte_atomic_32.h:12]: nice.

**Review by pawanpraka1 — APPROVED** [review]
looks good.

**Pull request merged**